flask_peewee/rest/__init__.py

```diff
@@ -770,6 +794,34 @@ def delete(self, obj):
         res = obj.delete_instance(recursive=self.delete_recursive)
         return self.response({'deleted': res})
 
+    def json_edit(self, obj, field_name, path):
+        data = self.read_request_data()
+        value = getattr(obj, field_name)
+        for key in path:
+            value = value.setdefault(key, {})
+
+        value.update(data)
+        obj.save()
+        return self.response(self.serialize_object(obj))
+
+    def json_delete(self, obj, field_name, path):
+        if not path:
+            return Response({'error': 'Cannot delete.'}, 403)
+
+        value = getattr(obj, field_name)
+        for key in path[:-1]:
+            value = value.setdefault(key, {})
+
+        key = path[-1]
+        deleted = False
+        if key in value:
+            del value[key]
+            deleted = True
+
+        obj.save()
```

Comment:
I think you need this

```suggestion
        if deleted:
            obj.save()
```

Otherwise if you have `foo.bar={}` and then `DELETE /api/foo/37/bar/baz/qux` you end up with `foo.bar={"baz": {}}`.

Reply:
Nice catch, will add a test for it as well.